_integration-test/custom-ca-roots/setup.sh

```diff
@@ -25,8 +25,11 @@ openssl req -new -nodes -newkey rsa:2048 -keyout $TEST_NGINX_CONF_PATH/self.test
 
 # openssl req -in nginx/self.test.req -text -noout
 
+# Store extension details in a temporary file since python subprocess can't use process substitution
+printf "subjectAltName=DNS:self.test" >/tmp/extension_details
```

Comment:
you should be able to do this inline in python just fine -- could you share what you're running into here?

as written this is susceptible to tempfile injection since it's using an easily guessable (static) path

Reply:
Ah, good point. This is the error I'm getting:
```
_integration-test/custom-ca-roots/setup.sh: line 29: syntax error near unexpected token `('
```

Noting that I do get the same thing running locally, but in CI it does not fail

Reply:
can you show what code led to that?

Reply:
Specifically this line:
```
openssl x509 -req -in $TEST_NGINX_CONF_PATH/self.test.req -CA $TEST_NGINX_CONF_PATH/ca.crt -CAkey $TEST_NGINX_CONF_PATH/ca.key \
  -extfile <(printf "subjectAltName=DNS:self.test") \
  -CAcreateserial -out $TEST_NGINX_CONF_PATH/self.test.crt -days 1 -sha256
```

the `<(printf "subjectAltName=DNS:self.test")`

Reply:
ah yeah so what's happening in bash is it's making a process and file descriptor and substituting that into the command

you can pretty easily do that in python as well -- but you need to manage the file descriptor yourself

it's probably easier to use `NamedTemporaryFile` rather than an anonymous descriptor

Reply:
Makes sense. I would like to eventually rewrite the entire file in python, but would like to do that in a follow up PR. For now, I decided to use `mktmp` so it's no longer a static path.